pkg/cloud/services/securitygroup/securitygroups_test.go

```diff
@@ -340,7 +341,7 @@ func TestReconcileSecurityGroups(t *testing.T) {
 			scheme := runtime.NewScheme()
 			_ = infrav1.AddToScheme(scheme)
 			client := fake.NewClientBuilder().WithScheme(scheme).Build()
-			scope, err := scope.NewClusterScope(scope.ClusterScopeParams{
+			cs, err := scope.NewClusterScope(scope.ClusterScopeParams{
```

Comment:
Since we are using `clusterscope` object, can we check the conditions set on `AWSCluster` object of `clusterscope` in success case?

Reply:
I think we could check this condition. But as per current PR scope, I have added the missing unit test cases and not focused on improving the existing test cases.
However to improve the existing test cases we could start a separate issue or can add it as part of adding integration tests(if required). Wdyt?